scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala

```diff
@@ -0,0 +1,87 @@
+package scala.meta.internal.parsers
+
+import scala.meta.Dialect
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Ident
+import scala.meta.internal.classifiers.classifier
+
+class SoftKeywords(dialect: Dialect) {
+
+  def isIdentAnd(token: Token, pred: String => Boolean): Boolean = token match {
+    case Ident(value) if pred(value.stripPrefix("`").stripSuffix("`")) => true
+    case _ => false
+  }
+
+  def isSoftKw(token: Token, name: String): Boolean = isIdentAnd(token, _ == name)
```

Comment:
Can this be `private`?

Reply:
also, could add dialect check as well, so it's always in the right order of checking:
```
  private matches(token: Token, name: String, isEnabled: Dialect => Boolean): Boolean =
    isEnabled(dialect) && isIdentAnd(token, _ == name)
```
(in this class, everything is about soft keywords, so i also renamed the method)

Reply:
Moved it actually to ScalametParserObject and is now reused in SoftKeywords and ScalametaParser.

Reply:
Ach no, not that one :sweat_smile: this one is renamed to `matches` and made private.